diff -urN a/td/generate/CMakeLists.txt b/td/generate/CMakeLists.txt
--- a/td/generate/CMakeLists.txt	2026-06-02 08:47:36.882241246 +0900
+++ b/td/generate/CMakeLists.txt	2026-06-02 08:48:28.413330277 +0900
@@ -176,6 +176,7 @@
 
   add_library(tl_writer_cpp STATIC ${TL_WRITER_CPP_SOURCE})
   target_link_libraries(tl_writer_cpp PRIVATE tdtl)
+  target_compile_options(tl_writer_cpp PRIVATE "-march=x86-64" "-O2")
   target_include_directories(tl_writer_cpp PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
   if (TD_ENABLE_JNI)
     target_compile_definitions(tl_writer_cpp PRIVATE TD_ENABLE_JNI=1 GIT_COMMIT_HASH=${TD_GIT_COMMIT_HASH})
@@ -186,6 +187,7 @@
 
   add_executable(generate_mtproto ${TL_GENERATE_MTPROTO_SOURCE})
   target_link_libraries(generate_mtproto PRIVATE tdtl tl_writer_cpp)
+  target_compile_options(generate_mtproto PRIVATE "-march=x86-64" "-O2")
   add_custom_target(tl_generate_mtproto
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/auto
     COMMAND generate_mtproto
@@ -195,6 +197,7 @@
 
   add_executable(generate_common ${TL_GENERATE_COMMON_SOURCE})
   target_link_libraries(generate_common PRIVATE tdtl tl_writer_cpp)
+  target_compile_options(generate_common PRIVATE "-march=x86-64" "-O2")
   add_custom_target(tl_generate_common
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/auto
     COMMAND ${GENERATE_COMMON_CMD}
@@ -207,6 +210,7 @@
 
   add_executable(generate_c ${TL_GENERATE_C_SOURCE})
   target_link_libraries(generate_c PRIVATE tdtl)
+  target_compile_options(generate_c PRIVATE "-march=x86-64" "-O2")
   add_custom_target(tl_generate_c
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/auto
     COMMAND generate_c
@@ -216,10 +220,12 @@
 
   add_executable(td_generate_java_api ${TL_GENERATE_JAVA_SOURCE})
   target_link_libraries(td_generate_java_api PRIVATE tdtl)
+  target_compile_options(td_generate_java_api PRIVATE "-march=x86-64" "-O2")
   target_compile_definitions(td_generate_java_api PRIVATE GIT_COMMIT_HASH=${TD_GIT_COMMIT_HASH})
 
   add_executable(generate_json ${TL_GENERATE_JSON_SOURCE})
   target_link_libraries(generate_json PRIVATE tdtl tdutils)
+  target_compile_options(generate_json PRIVATE "-march=x86-64" "-O2")
   add_custom_target(tl_generate_json
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/auto
     COMMAND generate_json
@@ -242,6 +248,7 @@
 
     add_executable(td_generate_dotnet_api generate_dotnet.cpp tl_writer_dotnet.h)
     target_link_libraries(td_generate_dotnet_api PRIVATE tdtl)
+    target_compile_options(td_generate_dotnet_api PRIVATE "-march=x86-64" "-O2")
     add_custom_target(generate_dotnet_api
       WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/auto
       COMMAND ${GENERATE_DOTNET_API_CMD}
diff -urN a/td/generate/tl-parser/CMakeLists.txt b/td/generate/tl-parser/CMakeLists.txt
--- a/td/generate/tl-parser/CMakeLists.txt	2026-06-02 08:47:36.874762598 +0900
+++ b/td/generate/tl-parser/CMakeLists.txt	2026-06-02 08:47:49.032880853 +0900
@@ -13,6 +13,7 @@
 endif()
 
 add_executable(${PROJECT_NAME} ${SOURCES})
+target_compile_options(${PROJECT_NAME} PRIVATE "-march=x86-64" "-O2")
 
 if (NOT WIN32)
   target_link_libraries(${PROJECT_NAME} PRIVATE m)
diff -urN a/tdtl/CMakeLists.txt b/tdtl/CMakeLists.txt
--- a/tdtl/CMakeLists.txt	2026-06-02 08:47:36.871313973 +0900
+++ b/tdtl/CMakeLists.txt	2026-06-02 08:47:44.166158441 +0900
@@ -25,4 +25,5 @@
 )
 
 add_library(tdtl STATIC ${TDTL_SOURCE})
+target_compile_options(tdtl PRIVATE "-march=x86-64" "-O2")
 target_include_directories(tdtl PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
diff -urN a/tdutils/generate/CMakeLists.txt b/tdutils/generate/CMakeLists.txt
--- a/tdutils/generate/CMakeLists.txt	2026-06-02 08:47:36.878551520 +0900
+++ b/tdutils/generate/CMakeLists.txt	2026-06-02 08:48:10.393125517 +0900
@@ -36,6 +36,7 @@
   set(GPERF_GEN_SOURCE generate_mime_types_gperf.cpp)
 
   add_executable(generate_mime_types_gperf ${GPERF_GEN_SOURCE})
+  target_compile_options(generate_mime_types_gperf PRIVATE "-march=x86-64" "-O2")
 
   add_custom_command(
     OUTPUT ${GPERF_FILES}
