diff --git a/PKGBUILD b/PKGBUILD
index 6de8709..114271e 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -22,9 +22,6 @@ b2sums=('6d2253f9840999a59b18d5ad9d1ec3f499a6ffcb16bc73a4f6c0643d85fc73cdc3129a5
 prepare() {
   cd $pkgname
   
-  # Fix double installation issue in Makefile.am files
-  patch -p1 -i ../byobu-6.14-fix-double-install.patch
-  
   # Adjust path to SOCKETDIR
   setconf etc/byobu/socketdir SOCKETDIR '"/tmp/screens"'
 
